src/dawn/SIR/SIR.h

```diff
@@ -314,9 +314,14 @@ struct Value : NonCopyable {
 
   template <class T>
   explicit Value(T&& value) : isConstexpr_(false) {
+    //TODO
+    //  I get a compiler error I don't understand when not using this redirection 
```

Comment:
I assume, in the end you will only have 
```
... : valueImpl(std::forward<T>(value)) {}
```
And the samme inside ValueImpl (and you can remove the pointer in ValueImpl). Maybe, this solves your compilre error too.

Reply:
Unfortunately valueImp_ has to be a pointer, see below

Reply:
Not the pointer to valueImpl (that's clear), but the pointer in valueImpl (ValuePtr)